### PR TITLE
initial template for dashboard

### PR DESCRIPTION
**Pull request by altairarclight**

### CODE REVIEW

File: pages/me.tsx [hunk 1]
```diff
@@ -4,35 +4,80 @@ import { supabase } from "../utils/client";
...
+//   <h1 className="text-3xl">Pinned Stocks</h1>
+//   <pre>
+//     {JSON.stringify(user, null, 2)}
+//     {user?.id}
+//   </pre>
+// </div>
+
 export default Dashboard;
+export const getServerSideProps = async context => {
+  withPageAuth({ redirectTo: '/login?from=/me' })()
```
Comment (Desdaemon) [inline pages/me.tsx:67]: This doesn't pass the `user` prop to the page, perhaps merge it in like this?
```jsx
import { withPageAuth } from '@supabase/auth-helpers-nextjs'

export default function ProtectedPage({ user, customProp }) {
  return <div>Protected content</div>
}

export const getServerSideProps = withPageAuth({
  redirectTo: '/foo',
  async getServerSideProps(ctx, supabase) {
    // Access the user object
    const {
      data: { user },
    } = await supabase.auth.getUser()
    return { props: { email: user?.email } }
  },
})
```

File: pages/me.tsx [hunk 2]
```diff
@@ -4,35 +4,80 @@ import { supabase } from "../utils/client";
...
+  async function getPinnedStocks() {
+    const {data, error} = await supabase
+    .from('Users')
+    .select()
+    .eq("user-id", "6ea9c27d-9b7b-4354-bad4-ad7d903492d4")
```
Comment (Desdaemon) [inline pages/me.tsx:72]: We should be given the user by the above page, could we use that instead of hard-coding the user-id?

**Review by Desdaemon — COMMENTED** [review]
LGTM

File: pages/me.tsx (hunk 1 above)
Comment (altairarclight) [reply pages/me.tsx]: Thanks. This workaround worked great.

File: pages/me.tsx (hunk 2 above)
Comment (altairarclight) [reply pages/me.tsx]: Yes, it is now added in the new commit

**Pull request merged**